### ISSUE TITLE
another oddity with names  (possibly from Panther pipeline)

### ISSUE THREAD

**Issue by ValWood** [geneontology/amigo#475]
another oddity with names  (possibly from Panther pipeline)

I notices whilst checking these
https://github.com/geneontology/helpdesk/issues/99

Sometimes when you search on a human name, you see a gene name "A3GALT2 (1 of many)"
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:H2UKJ8

These seem to be assigned by some GO pipeline because they don't come from the UniProt entry
http://www.uniprot.org/uniprot/H2UKJ8

Until genes are "officially" names by a naming authority, we really shouldn't transfer names. This one should have the unique currently assigned systematic name, which is 

Gene LOC101078261 (not very friendly but this is what we should use). We definitely should not be transferring human names to fish... and even more so one human name to multiple fish genes.

**Comment by kltm** [comment]
@vanaukenk @ukemi
I'm trying to find the right spot for this item; would it possibly in the go-annotation tracker, or maybe somewhere else to figure out the upstream source of the issue?

**Comment by vanaukenk** [comment]
@kltm 
The annotations for this gene seem to be coming from the PAINT pipeline, so perhaps the paint repo would be a reasonable place to start.

**Comment by ValWood** [comment]
This is quite a big issue for usability.

Column | Content | Required? | Cardinality | Example
3 | DB Object Symbol | required | 1 | PHO3

"A3GALT2 (1 of many)" might be added so that   the file conforms to the requirement to have a gene object symbol? (see above).

I'm guessing that it goes "A3GALT2 (1 of many)" , "A3GALT2 (2 of many)" etc.  to ensure cardinality "1". It is difficult to know because the  part of these labels in parentheses does not appear to be searchable in AmigGO search.

The rule of thumb is i) all genes should have a systematic name. ii) The systematic name is the "official name" until a symbol is assigned.

GO Docs also seem to agree with this:

DB Object Symbol (column 3)
a (unique and valid) symbol to which DB object ID is matched
**can use ORF name for otherwise unnamed gene or protein**

Then
if gene products are annotated, can use gene product symbol if available, or many gene product annotation entries can share a gene symbol this field is mandatory, cardinality 1
(although this  sentence does not make sense)

The DB Object Symbol field should be a symbol that means something to a biologist wherever possible (a gene symbol, for example). It is not an ID or an accession number (DB object ID [column 2] provides the unique identifier), although IDs can be used as a DB object symbol if there is no more biologically meaningful symbol available (e.g., when an unnamed gene is annotated).

ORF name is synonymous with "systematic name", == name assigned from the sequencing project (maybe this could be clearer).

GO should *never* assign symbols....

**Comment by pgaudet** [comment]
@dustine32 is this a ticket for the https://github.com/pantherdb/fullgo_paint_update/issues repo?

**Comment by ValWood** [comment]
This is definitely upstream of AMiGO, but I did not know where to report it because I wasn't sure where the problem originated.

I think this is bad enough to be flagged as a  "blocker" it really need to be addressed (affects usability)

**Comment by kltm** [comment]
As AmiGO will not be shut down in the interim, it's probably best not to use "showstopper".

**Comment by dustine32** [comment]
@pgaudet I can take a look at the examples and see if something needs to be fixed in the fullgo_paint_update pipelin, then create a ticket there if necessary. The "A3GALT2 (1 of many)" example seems different than the other Reference Proteome gene symbol problem we had here: https://github.com/geneontology/helpdesk/issues/112. Thanks!

**Comment by ValWood** [comment]
@dustine32  I think the basic problem here is that names are generated by the pipeline somehow.
This should *never* happen

**Comment by kltm** [comment]
@dustine32 Where should be move this ticket to?

**Comment by dustine32** [comment]
@kltm Still not sure. I don't think the "A3GALT2 (1 of many)" example can be fixed in the GO-PAINT update pipeline since this is the symbol in Panther's gene table (so thinking it's upstream at RefProt but need to confirm) and this table is only updated during the yearly "library building".

In short, I don't think the ticket should be moved to fullgo_paint_update.

**Comment by kltm** [comment]
@dustine32 A new home perhaps for these wayward tickets for PANTHER? Maybe https://github.com/pantherdb/Helpdesk or something new?

**Comment by dustine32** [comment]
@kltm Yes! That seems generic enough to hold this ticket for now. Thanks!

**Comment by cmungall** [comment]
I don't understand this ticket

This pufferfish protein has no name in either uniprot or GO. 

http://amigo.geneontology.org/amigo/gene_product/UniProtKB:H2UKJ8
https://www.uniprot.org/uniprot/H2UKJ8

These are consistent

Can someone clarify or close

**Comment by cmungall** [comment]
> Gene LOC101078261 (not very friendly but this is what we should use)

I'm perplexed, does this even exist?

https://www.uniprot.org/uniprot/?query=LOC101078261&sort=score

No results...

**Comment by dustine32** [comment]
@cmungall @ValWood Thanks to UniProt's handy compare tool, figured out that both of these were old names for this entry and have since been removed from UniProt:
2017-08-30
https://www.uniprot.org/uniprot/H2UKJ8?diff=true&version=22&version=25
`Name=A3GALT2 (1 of many)`

2018-02-28
https://www.uniprot.org/uniprot/H2UKJ8?diff=true&version=23&version=25
`Name=LOC101078261`

**Comment by ValWood** [comment]
Yes it seems the name was applied by an Ensembl automated pipeline.

- GN   Name=A3GALT2 (1 of many) {ECO:0000313|Ensembl:ENSTRUP00000037469};  

- CC   -!- CAUTION: The sequence shown here is derived from an Ensembl        
- CC       automatic analysis pipeline and should be considered as            
- CC       preliminary data. {ECO:0000313|Ensembl:ENSTRUP00000037469}.    


Analysis pipelines applying names based on orthology is  not a good idea....

In the current QuickGO https://www.ebi.ac.uk/QuickGO/annotations?geneProductId=H2UKJ8
symbol is H2UKJ8 , which will be confusing as most entries  do not have the Uniport accession as the name.

So now we have a situation where this entry is duplicated in AmiGO a with different "gene product" labels
See:
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:H2UKJ8

Maybe there should be some minimal criteria for a genome before the proteome can be included in AmiGO (not draft genome, unique systematic IDs applied, etc)  ?

**Comment by ValWood** [comment]
LOC101078261 is the systematic ID for this entry .
https://www.ncbi.nlm.nih.gov/gene/101078261

when I search Ensembl with LOC101078261
I get
 novel gene (Fugu Gene)
ENSTRUG00000014665 scaffold_56:1062627-1065945:-1
LOC101078261 (UniProtKB Gene Name) is an external reference matched to Gene ENSTRUG00000014665

LOC101078261 is clearly the original name assigned by  the sequence providers, but is stripped out by Ensembl and leaves the uniprot protein entry with no  systematic identifier.
This is a new level of crapness.....

I guess there is nothing GO can do except have stricter criteria for inclusion

**Comment by cmungall** [comment]
Thanks for tracing this down @ValWood and @dustine32 

Yes, I agree there is not much we can do here. This relates to a proposal that has been brewing that I only just explicitly materialized in a ticket:
https://github.com/geneontology/go-site/issues/756
(not that this would be a panacea, each database comes with its own problems).

Stricter inclusion criteria: I think this is worth discussing at the GO meeting. Do you want to put something in the agenda for it? My inclination is that if it's in the QfO reference protein set (which must be the case for any IBA) is that it is better to be inclusive, but perhaps have some kind of rating to show the users (this could be a new amigo feature request).

OK to close this now?

**Comment by pgaudet** [comment]
Please don't close. I am assigning myself to re-check. (I cannot monitor any more repos)
We can add some label to note that there is nothing to do for us but that the issue is known.

@dustine32 Can you give an estimates as to when this may be resolved? 

Thanks, Pascale

**Comment by ValWood** [comment]
So this is quite a big problem, even is we appreciate that the gene product in question here will eventually get the LOC101078261 ID in Uniprot.

There are other examples where panther is generating exact duplicate. Here is one:

![problem](https://user-images.githubusercontent.com/7359272/43885858-fae31430-9bb1-11e8-98fb-5e295a1307d0.jpg)

I appreciate that the entire Panther library cannot be generate more than once a year, but it would be very useful if gene names table  to consolidate names outside of this time frame. A year is a long time to wait for fixes which have such a huge effect on the data....

I will :
a) put the upstream problem of UniPRot entries excluding the locus ID on the Agenda, and let Maria know (No handle in the git Hub look up list), but she is attending the Montreal meeting.

b) Let Maria know

**Comment by ValWood** [comment]
I mailed Maria, but now I am baffled as to why this is in the reference proteome if it is an unreviewed entry.
https://www.uniprot.org/uniprot/H2UKJ8
Maybe I  don't understand the meaning of the term reference proteome....

**Comment by ValWood** [comment]
> We can add some label to note that there is nothing to do for us but that the issue is known.

we use a label  "waiting for external change"  which  works well because you can filter everything you are waiting on...

**Comment by dustine32** [comment]
@pgaudet I believe we're aiming to have the library built by this November.

@ValWood Regular (monthly?) updating of Panther's gene_symbol data might be feasible. Just not sure if this can be plugged into the monthly GO->PAINT update pipeline or, since the inputs would be results from querying UniProt, this would be a separate process entirely. I could also see the argument that this might over-complicate the process and introduce other issues. Could you open an issue [here](https://github.com/pantherdb/Helpdesk/issues) and we can figure out how feasible this is?

**Comment by kltm** [comment]
@pgaudet If you're needing track PANTHER issues, we should talk about the best way to triage them without cluttering unrelated trackers.

**Comment by cmungall** [comment]
I am still not convinced this is a PANTHER issue. PAINT, GO Central/AmiGO and QuickGO/GOA appear to all be consistent with the new example.

Nevertheless we should implement a system that guarantees consistency between what PANTHER submits and what EBI-GOA submits. This need to require changing the upstream PANTHER database. It can either be (1) in the code that runs at USC to populate the files that are submitted to GO (2) in the GO Central, when we merge. Note that we have already implemented 2 but the issue here is that we are only doing this for a subset of species at the moment.

Val has also asked an important question about what does reference proteome mean, any why we have duplicates for pufferfish, and why it includes unreviewed proteins. These are good questions we should improve our docs here. I too have been long confused by the different usages of the term "reference". See the google doc I made in 2015: https://docs.google.com/document/d/1l4hZw_ZTor8RoTjYTNS4FtGAba4HZrewmjccrSXJ9Hs/edit
